parking/backend/db/dbaccess.py

```diff
@@ -44,6 +46,11 @@ class DbAccess(object):
                                                p.location.longitude, p.price, p.capacity, 0)
         return park_id
 
+    async def get_parking_lot(self, park_id: int) -> Record:
+        async with self.pool.acquire() as conn:
+            parking_lot: Record = await conn.fetchrow(c.PARKINGLOTS_SELECT, park_id)
+        return parking_lot
+
     async def delete_parking_lot(self, park_id: int) -> Optional[int]:
```

Comment:
Can't say I like the return values of these functions. You'd expect a success or failure if you were trying to delete something, but instead you're getting either some random ID or a null value? Not only are the return types different but they also seem pretty irrelevant.

Reply:
This has to do with consistency. Since insert_parking_lot needs to return an id or nothing, we chose all other updates/deletes to return the same thing. I agree that the alternative would be true/false.

Reply:
Then my argument would be that this is consistently bad practice!! Wouldn't it be better to return a tuple or something with the status and the relevant data? e.g. (success = True, data = new_id) or (success = false, data = null)

Reply:
Usually, in python, invalid states are flagged by raising exceptions.

If we assume nobody will try delete a parking lot that doesn't exist, then this is clearly an error and should raise an exception.

Either way, it's better to be consistent about these things than for some to be one way in one place and others to be a different way in other places.

Reply:
This is again a delicate balance. For something like `delete` there are a few possibilities:

- The delete succeeds: The callee requires no further information (other than the coroutine returns)
- The delete fails because it didn't exist: The callee needs to be informed that it didn't exist
- The delete fails because of a general DB error (e.g. query took too long): The callee needs to be informed of DB errors

For the first possibility the function can just return nothing (the id is meaningless, especially as the entry no longer exists). For the third possibility we can consider this as a firmly "exceptional" circumstance so an exception should definitely be raised (which I imagine asyncpg will handle for us).

The second possibility is more nuanced. The id of the lot being deleted does not come from within the program but from external input therefore we cannot say it is "exceptional" or an error for this to occur, for instance two deletes for the same lot may come in at the same time, so one might be inclined to use a boolean return value to indicate failure. That being said the "exceptional" criteria does not apply to Python. Exceptions _are_ recommended for control flow in pythonic code (within reason) and I think in this instance would be suitable as we ultimately need to propagate this up as a 404 response.

This is basically a longwinded way of saying I agree with Ben.